lib/registry/src/graphql.rs

```diff
@@ -0,0 +1,251 @@
+    /// If a proxy is specified in wapm config's `proxy.url`, it is assumed
+    /// to be a general proxy
+    ///
+    /// A return value of `Ok(None)` means that there was no attempt to set up a proxy,
+    /// `Ok(Some(proxy))` means that the proxy was set up successfully, and `Err(e)` that
+    /// there was a failure while attempting to set up the proxy.
+    pub fn maybe_set_up_proxy() -> anyhow::Result<Option<reqwest::Proxy>> {
+        use std::env;
+        let proxy = if let Ok(proxy_url) = env::var("ALL_PROXY").or_else(|_| env::var("all_proxy"))
+        {
+            reqwest::Proxy::all(&proxy_url).map(|proxy| (proxy_url, proxy, "ALL_PROXY"))
+        } else if let Ok(https_proxy_url) =
+            env::var("HTTPS_PROXY").or_else(|_| env::var("https_proxy"))
+        {
+            reqwest::Proxy::https(&https_proxy_url)
+                .map(|proxy| (https_proxy_url, proxy, "HTTPS_PROXY"))
+        } else if let Ok(http_proxy_url) =
+            env::var("HTTP_PROXY").or_else(|_| env::var("http_proxy"))
+        {
+            reqwest::Proxy::http(&http_proxy_url).map(|proxy| (http_proxy_url, proxy, "http_proxy"))
+        } else {
+            return Ok(None);
+        }
+        .map_err(|e| ProxyError::ConnectionError(e.to_string()))
+        .and_then(
+            |(proxy_url_str, proxy, url_location): (String, _, &'static str)| {
+                url::Url::parse(&proxy_url_str)
+                    .map_err(|e| ProxyError::UrlParseError {
+                        url_location: url_location.to_string(),
+                        error_message: e.to_string(),
+                    })
+                    .map(|url| {
+                        if !(url.username().is_empty()) && url.password().is_some() {
+                            proxy.basic_auth(url.username(), url.password().unwrap_or_default())
+                        } else {
+                            proxy
+                        }
+                    })
+            },
+        )?;
+
+        Ok(Some(proxy))
+    }
+}
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    schema_path = "graphql/schema.graphql",
+    query_path = "graphql/queries/get_package_version.graphql",
+    response_derives = "Debug"
+)]
+pub(crate) struct GetPackageVersionQuery;
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    schema_path = "graphql/schema.graphql",
+    query_path = "graphql/queries/get_package_by_command.graphql",
+    response_derives = "Debug"
+)]
+pub(crate) struct GetPackageByCommandQuery;
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    schema_path = "graphql/schema.graphql",
+    query_path = "graphql/queries/test_if_registry_present.graphql",
+    response_derives = "Debug"
+)]
+pub(crate) struct TestIfRegistryPresent;
+
+#[cfg(target_os = "wasi")]
+pub fn whoami_distro() -> String {
+    whoami::os().to_lowercase()
+}
+
+#[cfg(not(target_os = "wasi"))]
+pub fn whoami_distro() -> String {
+    whoami::distro().to_lowercase()
+}
+
+pub fn execute_query_modifier_inner_check_json<V, F>(
```

Comment:
I don't think we need this funciton. Where and why is used?

Reply:
It is being used in the `test_if_registry_present` function. I've added a comment.